File: 04_quicksort/javascript/quick_sort/06_quicksort-gus.js
const quickSort = (array) => {
  if (array.length <= 1) {
    return array;
  }

  let pivot = array[0];
  let menores = [];
  let maiores = [];

  for (let index = 1; index < array.length; index += 1) {
    if (array[index] < pivot) {
      menores.push(array[index]);
    } else {
      maiores.push(array[index]);
    }
  }

  return [...quickSort(menores), pivot, ...quickSort(maiores)];
};

File: 04_quicksort/python/06_quicksort-gus.py
import random

def quicksort(array):
    if len(array) < 2:
        return array  # Caso base: listas com 0 ou 1 elemento já estão ordenadas
    else:
        pivot_index = random.randint(0, len(array) - 1)  # Escolhe um pivô aleatório
        pivot = array[pivot_index]
        
        # Particionamento dos elementos em relação ao pivô
        less = [i for i in array if i < pivot]
        equal = [i for i in array if i == pivot]
        greater = [i for i in array if i > pivot]
        
        return quicksort(less) + equal + quicksort(greater)

# Exemplo de uso
print(quicksort([10, 5, 2, 3]))
